packages/@aws-cdk/aws-codedeploy/lib/ecs/deployment-group.ts

```diff
@@ -32,14 +46,144 @@ export interface IEcsDeploymentGroup extends cdk.IResource {
 }
 
 /**
- * Note: This class currently stands as a namespaced container for importing an ECS
- * Deployment Group defined outside the CDK app until CloudFormation supports provisioning
- * ECS Deployment Groups. Until then it is closed (private constructor) and does not
- * extend {@link Construct}.
+ * Details about an ECS service and its corresponding ECS cluster
+ */
+export type EcsService = ecs.Ec2Service | ecs.FargateService | ecs.ExternalService;
+
+/**
+ * A route with an ELB listener and target group.
+ */
+export interface EcsTrafficRoute {
+  /**
+   * The load balancer for the traffic route.
+   *
+   * @default - blank if empty on test traffic route. Required for prod traffic route.
+   */
+  readonly listener: elbv2.IApplicationListener | elbv2.INetworkListener | undefined;
+  /**
+   * The target group for the traffic route.
+   */
+  readonly targetGroup: elbv2.ITargetGroup;
+}
+
+/**
+ * Configuration for the blue/green CodeDeploy deployment.
+ */
+export interface EcsBlueGreenDeploymentConfig {
+  /**
+   * The time to wait for a ContinueDeployment after the deployment is ready.
+   * Leave blank to automatically continue once the deployment is ready.
+   *
+   * @default - automatically continue once deployment is ready
+   */
+  readonly waitTimeForContinueDeployment?: cdk.Duration;
+  /**
+   * The time to wait before terminating old tasks.
+   * Leave blank to keep the old tasks alive and deregistered from the load balancer.
+   *
+   * @default - keep the old tasks alive and deregistered from the load balancer.
+   */
+  readonly waitTimeForTermination?: cdk.Duration;
+}
+
+/**
+ * Construction properties for {@link EcsDeploymentGroup}.
+ */
+export interface EcsDeploymentGroupProps {
+  /**
+   * The reference to the ECS Services to deploy.
+   *
+   * [disable-awslint:ref-via-interface] because the service needs to have the deploy controller changed to Code Deploy
+   */
+  readonly services: EcsService[];
+
+  /**
+   * The reference to the production traffic group for the deployment.
+   * Leave blank to disable traffic shifting
+   *
+   * @default - Traffic shifting will be disabled
+   */
+  readonly prodTrafficRoute?: EcsTrafficRoute;
+
+  /**
+   * The reference to the test traffic group for the deployment.
+   *
+   * @default - One will be created for you
+   */
+  readonly testTrafficRoute?: EcsTrafficRoute;
```

Comment:
For ECS deployments, CodeDeploy requires 2 target groups (one blue, one green), regardless of whether there's a test route.

Reply:
I was trying to auto-create the green TG as a clone of blue TG. More trouble than its worth - i've abandon that idea.